src/core/AudioEngine.cpp

```diff
@@ -808,57 +803,42 @@ void AudioEngine::removePlayHandlesOfTypes(Track * track, PlayHandle::Types type
 
 void AudioEngine::requestChangeInModel()
 {
-	if( s_renderingThread )
-		return;
+	if (s_renderingThread || !m_isProcessing) { return; }
 
-	m_changesMutex.lock();
-	m_changes++;
-	m_changesMutex.unlock();
-
-	m_doChangesMutex.lock();
-	m_waitChangesMutex.lock();
-	if (m_isProcessing && !m_waitingForWrite && !m_changesSignal)
 	{
-		m_changesSignal = true;
-		m_changesRequestCondition.wait( &m_waitChangesMutex );
+		auto lock = std::lock_guard{m_numChangesMutex};
+		++m_numChanges;
```

Comment:
> If you're trying to simplify code, m_numChanges could be atomic, which would mean it no longer needs a mutex to guard access to it. This would also reduce the number of mutexes used in real-time code (any code blocking a real-time thread, such as code making changes to the model, should itself be considered real-time code).

Making ``m_numChanges`` atomic was my intention at first, but I believe with the use of ``std::condition_variable`` it could cause things to be out the sync. 

From [cppreference](https://en.cppreference.com/w/cpp/thread/condition_variable):
``
Even if the shared variable is atomic, it must be modified while owning the mutex to correctly publish the modification to the waiting thread. 
``